src/tests/tracing/eventpipe/dynamicallocationsampling/manual/Program.cs

```diff
@@ -0,0 +1,435 @@
+        private static int _allocationsCount = 0;
+        private static List<string> _allocatedTypes = new List<string>();
+
+        static void Main(string[] args)
+        {
+            if (args.Length == 0)
+            {
+                Console.WriteLine("No process ID specified");
+                return;
+            }
+
+            int pid = -1;
+            if (!int.TryParse(args[0], out pid))
+            {
+                Console.WriteLine($"Invalid specified process ID '{args[0]}'");
+                return;
+            }
+
+            try
+            {
+                PrintEventsLive(pid);
+            }
+            catch (Exception x)
+            {
+                Console.WriteLine(x.Message);
+            }
+        }
+
+
+        public static void PrintEventsLive(int processId)
+        {
+            var providers = new List<EventPipeProvider>()
+            {
+                new EventPipeProvider(
+                        "Microsoft-Windows-DotNETRuntime",
+                        EventLevel.Verbose, // verbose is required for AllocationTick
+                        (long)0x80000000001 // new AllocationSamplingKeyword + GCKeyword
+                        ),
+                new EventPipeProvider(
+                        "Allocations-Run",
+                        EventLevel.Informational
+                        ),
+            };
+            var client = new DiagnosticsClient(processId);
+
+            using (var session = client.StartEventPipeSession(providers, false))
+            {
+                Console.WriteLine();
+
+                Task streamTask = Task.Run(() =>
+                {
+                    var source = new EventPipeEventSource(session.EventStream);
+                    ClrTraceEventParser clrParser = new ClrTraceEventParser(source);
+                    clrParser.GCAllocationTick += OnAllocationTick;
+                    source.Dynamic.All += OnEvents;
+
+                    try
+                    {
+                        source.Process();
+                    }
+                    catch (Exception e)
+                    {
+                        Console.WriteLine($"Error encountered while processing events: {e.Message}");
+                    }
+                });
+
+                Task inputTask = Task.Run(() =>
+                {
+                    while (Console.ReadKey().Key != ConsoleKey.Enter)
+                    {
+                        Thread.Sleep(100);
+                    }
+                    session.Stop();
+                });
+
+                Task.WaitAny(streamTask, inputTask);
+            }
+
+            // not all cases are emitting allocations run events
+            if ((_sampledTypesInRun == null) && (_sampledTypes.Count > 0))
+            {
+                ShowIterationResults();
+            }
+        }
+
+        private const long SAMPLING_MEAN = 100 * 1024;
+        private const double SAMPLING_RATIO = 0.99999 / 0.00001;
```

Comment:
There was a 2nd bug affecting the stats here. p and q were set assuming that we were sampling once every 100,000 bytes, but the runtime implementation is sampling once every 102400 bytes causing a 2.4% bias. We'll need to correct all mentions of 'p' and 'q' in the docs to be 1/102400 and 1 - 1/102400.

Reply:
I'm changing the upscaling code and general references in doc
Don't you mind changing the numbers you computed in section "Including error bounds in the estimate" of \docs\design\features\DynamicAllocationSampling.md?

Reply:
Sure, I'll change it.